packages/components/form-field/src/FormFieldStateMessage.tsx

```diff
@@ -0,0 +1,34 @@
+import { forwardRef } from 'react'
+
+import { useFormField } from './FormFieldContext'
+import { FormFieldMessage, FormFieldMessageProps } from './FormFieldMessage'
+import {
+  formFieldStateMessageStyles,
+  FormFieldStateMessageStylesProps,
+} from './FormFieldStateMessage.styles'
+
+export interface FormFieldStateMessageProps
+  extends FormFieldMessageProps,
+    FormFieldStateMessageStylesProps {}
+
+export const FormFieldStateMessage = forwardRef<HTMLSpanElement, FormFieldStateMessageProps>(
+  ({ className, state, ...others }, ref) => {
+    const field = useFormField()
+
+    if (field.state !== state) {
+      return null
```

Comment:
If I understand correctly the point here would be to avoid incoherent states? But why having to declare a state on this component, if it relies on FormField context? :thinking:

Reply:
I was thinking about a third option, I guess:
```tsx
<FormField status="error">
  <Input />
  <FormField.StatusMessage status={status.token}>{status.message}</FormField.StatusMessage>
</FormField>
```

Reply:
Well I was thinking more about the second option, but maybe we also could consider another one:
```tsx
const defaultStatusMessages = {
 error: 'Something went wrong',
  success: 'Great! Message sent!',
  alert: 'Warning, youre house is under fire...',
}

<FormField status="error" statusMessages={defaultStatusMessages}>
  <Input />
  <FormField.StatusMessage />
</FormField>
```
...where the message itself would be extracted from `defaultStatusMessages` (this prop could have default value for sake of simplicity). Wdyt?

Reply:
Agree 👍🏻 I think that we can keep both APIs but in the examples only illustrate the second one. I added the three components perhaps tests and example could be added in another PR